core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,179 @@
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.story instanceof StorySummary) {
+      this.setStorySummary(this.story);
+    } else if (this.story instanceof CollectionSummary) {
+      this.setCollectionSummary(this.story);
+    } else {
+      this.setExplorationSummary(this.story);
+    }
+  }
+
+  setStorySummary(storyModel: StorySummary): void {
+    const completedStories = storyModel.getCompletedNodeTitles().length;
+    this.desc = storyModel.getTitle();
+    this.imgColor = storyModel.getThumbnailBgColor();
+    this.imgUrl = this.getStorySummaryThumbnailUrl({
+      filename: storyModel.getThumbnailFilename(),
+      id: storyModel.getId(),
+    });
+
+    const nextStory =
+      completedStories === storyModel.getAllNodes().length
+        ? completedStories - 1
+        : completedStories;
+
+    const lessonArgs = {
+      classroom: storyModel.getClassroomUrlFragment(),
+      topicFragment: storyModel.getTopicUrlFragment(),
+      currentStory: storyModel.getAllNodes()[nextStory],
+      story: storyModel.getUrlFragment(),
+    };
+    this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+    this.title = `Chapter ${nextStory + 1}: ${storyModel.getNodeTitles()[nextStory]}`;
+    this.progress = Math.floor(
+      (completedStories / storyModel.getNodeTitles().length) * 100
+    );
+    this.lessonTopic = this.topic;
+  }
+
+  setCollectionSummary(collectionModel: CollectionSummary): void {
+    this.desc = collectionModel.objective;
+    this.imgColor = collectionModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      collectionModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = collectionModel.title;
+    this.lessonUrl = `/collection/${collectionModel.id}`;
+    this.lessonTopic = 'Collections';
+  }
+
+  setExplorationSummary(explorationModel: LearnerExplorationSummary): void {
+    this.desc = explorationModel.objective;
+    this.imgColor = explorationModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      explorationModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = explorationModel.title;
+    this.lessonUrl = `/explore/${explorationModel.id}`;
+    this.lessonTopic = 'Community Lessons';
+  }
+
+  getStorySummaryThumbnailUrl({filename, id}: ThumbnailUrl): string {
+    if (!filename) {
+      return this.urlInterpolationService.getStaticImageUrl(
+        '/subjects/Lightbulb.svg'
+      );
+    }
+    return this.assetsBackendApiService.getThumbnailUrlForPreview(
+      AppConstants.ENTITY_TYPE.STORY,
+      id,
+      filename
+    );
+  }
+
+  getStorySummaryLessonUrl({
+    classroom,
+    topicFragment,
+    story,
+    currentStory,
+  }: LessonUrl): string {
+    return classroom === undefined || topicFragment === undefined
```

Comment:
That's a complicated ternary operator, I suggest replacing it with a simple
if (...)
   return '#'
return <generic_code>

And again, is classroom being undefined, or topicFragment being undefined is a normal situation? Shouldn't this throw an exception?

Reply:
I was just following the old code with this, but them being undefined doesn't seem possible. They have to be declared when creating them. Removed if statement.

Reply:
Ah, I realized the StorySummary model returns string | undefined for both classroom and topic fragments. I put back in the if statement for now.